src/nvim/screen.c

```diff
@@ -1314,12 +1314,14 @@ static void win_update(win_T *wp, Providers *providers)
   for (size_t k = 0; k < kv_size(*providers); k++) {
     DecorProvider *p = kv_A(*providers, k);
     if (p && p->redraw_win != LUA_NOREF) {
-      FIXED_TEMP_ARRAY(args, 4);
+      FIXED_TEMP_ARRAY(args, 6);
       args.items[0] = WINDOW_OBJ(wp->handle);
       args.items[1] = BUFFER_OBJ(buf->handle);
       // TODO(bfredl): we are not using this, but should be first drawn line?
       args.items[2] = INTEGER_OBJ(wp->w_topline-1);
       args.items[3] = INTEGER_OBJ(knownmax);
+      args.items[4] = INTEGER_OBJ(wp->w_leftcol);
+      args.items[5] = INTEGER_OBJ(wp->w_leftcol + wp->w_width + 1);
```

Comment:
let's exclude this. the end column is harder to calculate and will depend on the line (so if we do it later on, it is gonna be in the "line" event).

Reply:
Exclude the end column you mean ?